haskell-tests/Types/CBOR.hs

```diff
@@ -301,11 +302,231 @@ testEncodedTokenOperations = describe "EncodedTokenOperations JSON serialization
                     invalidEncTops1
             )
 
+emptyStringHash :: Hash.Hash
+emptyStringHash = Hash.Hash (FBS.pack [0xe3, 0xb0, 0xc4, 0x42, 0x98, 0xfc, 0x1c, 0x14, 0x9a, 0xfb, 0xf4, 0xc8, 0x99, 0x6f, 0xb9, 0x24, 0x27, 0xae, 0x41, 0xe4, 0x64, 0x9b, 0x93, 0x4c, 0xa4, 0x95, 0x99, 0x1b, 0x78, 0x52, 0xb8, 0x55])
+
+testTokenModuleStateSimpleJSON :: Spec
+testTokenModuleStateSimpleJSON = describe "TokenModuleState JSON serialization without additional state" $ do
+    let object =
```

Comment:
I'd prefer a randomized quickcheck test for the (de)-serialization tests.

Reply:
I agree. Use property-based testing similar to how it is done with the CBOR encoding:
```
    it "Encode and decode TokenModuleState (simple)" $ withMaxSuccess 1000 $ forAll genTokenModuleStateSimple $ \tt ->
        (Right ("", tt))
            === ( deserialiseFromBytes
                    decodeTokenModuleState
                    (toLazyByteString $ encodeTokenModuleState tt)
                )
```
For some additional regression tests, I would suggest having one or two known `TokenModuleState` values and their corresponding (textual) JSON representations and check that the encoding/decoding is correct.